go-runtime/compile/schema.go

```diff
@@ -548,6 +549,78 @@ func maybeVisitTypeEnumVariant(pctx *parseContext, node *ast.GenDecl, directives
 	return false
 }
 
+func visitTypeValue(pctx *parseContext, named *types.Named, tnode ast.Expr, index types.Type, isExported bool) optional.Option[*schema.TypeValue] {
```

Comment:
Maybe a comment about what this functions purpose is? I also don't quite recall what the purpose of `TypeValue` is, so a comment on that type would be nice.